backend-lib/src/main/rust/lib.rs

```diff
@@ -2169,8 +2157,7 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_requiresS
     let res = catch_unwind(&mut env, |env| {
         let _span = tracing::info_span!("RustBackend.pcztRequiresSaplingProofs").entered();
 
-        let pczt = Pczt::parse(&env.convert_byte_array(pczt)?[..])
-            .map_err(|e| anyhow!("Invalid PCZT: {:?}", e))?;
+        let pczt = parse_pczt(env, &pczt)?;
 
         let prover = Prover::new(pczt);
 
```

Comment:
Aside (just below): I'm perplexed that there is no `From<bool> for jboolean`. Doesn't matter I guess; this pattern only occurs once.